src/dredd.coffee

```diff
@@ -19,22 +19,33 @@ configureReporters = require './configure-reporters'
 
 CONNECTION_ERRORS = ['ECONNRESET', 'ENOTFOUND', 'ESOCKETTIMEDOUT', 'ETIMEDOUT', 'ECONNREFUSED', 'EHOSTUNREACH', 'EPIPE']
 
+removeDuplicates = (arr) ->
+  arr.reduce (alreadyProcessed, currentItem) ->
+    if alreadyProcessed.indexOf(currentItem) is -1
+      return alreadyProcessed.concat currentItem
+    return alreadyProcessed
+  , []
+
 class Dredd
-  constructor: (configOrigin) ->
+  constructor: (configOrigin, customConfig) ->
```

Comment:
Why changing/extending the interface here? I thought custom configuration will be maintained under `configOrigin['options']['custom']` and it can handle an object when using JS API directly and it will contain an array of encoded key-value pairs if initiated via CLI `--custom` interface.